exercises/tom_hampson/ex02_oo_basics/src/ShapeSorter.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include "Shape.h"
+
+#include <vector>
+#include <memory>
+
+/**
+ * @brief "Shape sorter" class that stores a vector of shapes and contains
+ * a few functions for printing the shapes.
+*/
+class ShapeSorter
+{
+public:
+  ShapeSorter() {};
+  virtual ~ShapeSorter();
+
+  void printShapesOfType(const std::string& shapeType) const;
+  void printShapesWithNSides(const int n) const;
+  void printShapesAreaDescending() const;
+  void printShapesPerimeterDescending() const;
+  
+  void addShape(std::shared_ptr<Shape>);
+  void removeShape(std::shared_ptr<Shape>);
+
+  static bool compareArea(std::shared_ptr<const Shape> shapeA, std::shared_ptr<const Shape> shapeB);
```

Comment:
Arguments to these functions could be passed by reference, unless we are trying to shared ownership of each shape with the function. Maybe passing each shape by a pointer/reference would be better, as then we are leaving the ownership of the objects to the caller.
```suggestion
  static bool compareArea(Shape * shapeA, Shape *shapeB);
```
Same for the comparePerimeter function

Reply:
Changed to reference